### PR TITLE
Nixify the project.

### PR DESCRIPTION
**Pull request by bitbloxhub**
Add a nix developer environment. WIP!

### CODE REVIEW

**Review by bitbloxhub — COMMENTED** [review]
Right now, I am running into an issue similar to [this](https://discourse.nixos.org/t/python-importerror-libstdc-so-6-cannot-open-shared-object-file-no-such-file-or-directory/36988), where a python package is trying to get `libstdc++.so.6` and cannot find it. If i set `LD_LIBRARY_PATH` as reccomended in [here](https://discourse.nixos.org/t/python-importerror-libstdc-so-6-cannot-open-shared-object-file-no-such-file-or-directory/36988/4), then i run into an issue similar to [this](https://discourse.nixos.org/t/new-install-nix-wants-glibc-2-38-debian-12-has-2-36/36109), where it appears to be trying to use the system glibc.

**Comment by billkamp** [comment]
Fyi Here is the current j2do list

GPB Jonah To Dos's

 

TODO:

 

https://annas-archive.org/

https://github.com/aigis-llm for Jonah AI

Using port 8069-8092

https://localai.io/advanced

frm bills phone

nix-flake-update

git hub actions 

Nix-collect-garbage

Nix home manager 

Catppuccin

https://nix.dev/tutorials/nix-language.html 

nixos-config

atuin for nushell

atuin sync as well

carapace and starship as well

update all should have for nix:

----------------------

cd /etc/nixos

git pull

sudo nixos-rebuild switch

------------------------------------

TODOs for GPB and Jonah:

rpi nix install (zoom)

kupd nix (you have this at the bottom of your notes)

nix-on-droid setup (zoom)

other fun nix things (zoom?)

collaborating on aigis (github pull requests from branches on aigis and probably some zoom meetings)

WSL2 with arch and gui  for 4H. Yes I have fractional scaling. 

Remote access to the vm via rdp:

VBoxManage modifyvm VM-name --vrde on

For kryz:

Step 0: Use KDE Plasma 6. Done.

Step 1: Set up home-manager on it so we can use Floorp and Nushell. We need to zoom for this.

Step 2: Switch home-manager to use their Floorp config options to install extensions and stuff.

Step 3: Probably more nix.

Block Chain

look at ethereum and ipfs, they are important. learn solidity and truffle 

development framework and ganache mock blockchain

use geth not meatamask

use light sync

 

Nix Remote Access ???:

From Perplexity: 

To remotely access a newly installed NixOS desktop, there are several options you can consider, each with its own setup process and features. Here are some of the best methods:

XRDP:

XRDP is supported natively on NixOS and allows remote desktop access using the RDP protocol, which is compatible with Windows clients.

To set up XRDP, add the following to your configuration.nix:

text

services.xserver.enable = true;

services.xserver.displayManager.sddm.enable = true; # or another display manager

services.xserver.desktopManager.plasma5.enable = true; # or another desktop manager

services.xrdp.enable = true;

services.xrdp.defaultWindowManager = "startplasma-x11"; # or another window manager

services.xrdp.openFirewall = true;

 

Rebuild your system with sudo nixos-rebuild switch to apply the changes

 

Done:

 

Bill Kamp

IAGP

(651) 308-3703

***@***.***

**Comment by billkamp** [comment]
11/14/24: When I run git pull in /etc/nixos, I get:

error: insufficient permission for adding an object to repository database .git/objects

fatal: failed to write object

fatal: unpack-objects failed

 

Bill Kamp

IAGP

(651) 308-3703

***@***.***

**Comment by bitbloxhub** [comment]
i forgot to add `--bun` to bunx, so it was trying to run node and not bun 🤦🏻. also i wasted some time misspelling library when setting `LD_LIBRARY_PATH`.

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                  libssh
+                  bzip2
+                  libxml2
+                  acl
+                  libsodium
+                  util-linux
+                  xz
+                  systemd
+                ]
+              );
```
Comment (bitbloxhub) [inline flake.nix:57]: I need to replace this with a wrapped python and add documentation on how to register it with rye.

**Review by billkamp — APPROVED** [review]
Looks good, wish I knew what it meant.

**Pull request merged**